lerna-config.d.ts

```diff
@@ -0,0 +1,101 @@
+/** lerna.json config file */
+export interface LernaConfig extends RootAndPerCommandOptions, IgnoreChangesOption {
```

Comment:
This is very interesting, but I'm a little leery of an external declaration file separate from the code that defines these options. Is it possible to transform Yargs config into type declarations, for instance? That would go a long way toward making this more maintainable.

Reply:
Ideally yargs comes up with a way to declare yargs config as decorators on an interface. (on a class)  They're finally porting to TypeScript so they might do something like this.

Reply:
Yes, that would be nice.

Reply:
If I wanted to push this over the finish line, it seems like this might be the only hold up.

I definitely get not wanting to lose parity between code and this typing... Is there file or folder of examples I could run a type-check against to test for parity?